atom/browser/net/atom_network_delegate.cc

```diff
@@ -241,6 +243,12 @@ int AtomNetworkDelegate::OnBeforeSendHeaders(
     net::URLRequest* request,
     const net::CompletionCallback& callback,
     net::HttpRequestHeaders* headers) {
+  auto cmd_line = base::CommandLine::ForCurrentProcess();
+  auto client_id = cmd_line->GetSwitchValueASCII(
+      switches::kDevToolsEmulateNetworkConditionsClientId);
+  if (!client_id.empty())
+    headers->SetHeader(
+        switches::kDevToolsEmulateNetworkConditionsClientId, client_id);
```

Comment:
The header is added only when emulation is enabled, chromium adds the header whenever a devtools client is attached.